test/Microsoft.NET.Sdk.Razor.Tests/StaticWebAssets/OverrideHtmlAssetPlaceholdersTest.cs

```diff
@@ -0,0 +1,245 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable disable
+
+using Microsoft.AspNetCore.StaticWebAssets.Tasks;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.AspNetCore.Razor.Tasks;
+
+public class OverrideHtmlAssetPlaceholdersTest
+{
+    [Theory]
+    [InlineData(
+        """
+        <script src="main#[.{fingerprint}].js"></script>
+        """,
+        true,
+        "main.js"
+    )]
+    [InlineData(
+        """
+        <script src="main#[.{fingerprint}].js">
+        </script>
+        """,
+        true,
+        "main.js"
+    )]
+    [InlineData(
+        """
+        <script    src="main#[.{fingerprint}].js"   >   </script>
+        """,
+        true,
+        "main.js"
+    )]
+    [InlineData(
+        """
+        <script src="./main#[.{fingerprint}].js"></script>
+        """,
+        true,
+        "./main.js"
+    )]
+    [InlineData(
+        """
+        <script src="./folder/folder/file.name.something#[.{fingerprint}].js"></script>
+        """,
+        true,
+        "./folder/folder/file.name.something.js"
+    )]
+    [InlineData(
+        """
+        <script src="main#[.{fingerprint}].suffix.js"></script>
+        """,
+        true,
+        "main.suffix.js"
+    )]
+    [InlineData(
+        """
+        <script src="/root/main#[.{fingerprint}].suffix.js"></script>
+        """,
+        true,
+        "/root/main.suffix.js"
+    )]
+    [InlineData(
+        """
+        <script src="main.js"></script>
+        """,
+        false
+    )]
+    [InlineData(
+        """
+        <script src='main#[.{fingerprint}].js'></script>
+        """,
+        false
+    )]
+    [InlineData(
+        """
+        <script src=main#[.{fingerprint}].js></script>
+        """,
+        false
+    )]
+    [InlineData(
+        """
+        <link href="main#[.{fingerprint}].js" rel="preload" as="script" fetchpriority="high" crossorigin="anonymous">
+        """,
+        true,
+        "main.js"
+    )]
+    public void ValidateAssetsRegex(string input, bool shouldMatch, string fileName = null)
```

Comment:
This made me think of something. How much do we care about users writing things like
```
<link
  href="..."
/>
```
Which afaik, is perfectly valid.

Reply:
It's valid and it works thanks to `\s+`. Added more tests which uncovered I missed one `+`. Thank you!